### DISCUSSION TITLE
How to add permissions to build project role like task role?

### DISCUSSION THREAD

**Discussion by joshgoodson** [aws/copilot-cli#2535] [Q&A] [answered]
It has come in handy trying to add additional policies to the task role for my services, but I'm curious if I can do this today with the build project role?

For example, I have internal package repositories (CodeArtifact, or other self-hosted repository) I'd like to add policy permissions for to the build project role, but it doesn't appear like the same mechanism to apply addon managed policies to the task role exist for the build project role.

Is this possible today or should I file a feature request?

**Comment by Lou1415926** [comment, marked as answer]
Hi @joshgoodson !

Unfortunately there is no built-in way of adding additional permissions to your build project role as of now. Please feel free to file a feature request!

You could attach a policy or modify the code build policy manually through console or aws cli, but that wouldn't be ideal since next time you deploy the pipeline, the role policies template will be overwritten by our generated one. This could be a temporary workaround though!

**Reply by Lou1415926** [reply]
@joshgoodson  Sorry I think I've given you an inaccurate answer.

You can modify the code build policy through console / aws cli, as long as you are not updating it using CFN template. That way, next time you run `copilot pipeline update`, the template will be overwritten.

As long as you modify the resource itself (for example, through IAM console), then the modification should be kept intact.

**Reply by joshgoodson** [reply]
Feature request: https://github.com/aws/copilot-cli/issues/2755
